cmd/nodecmd/create.go

```diff
@@ -0,0 +1,678 @@
+	awsInstanceBody.SetAttributeValue("instance_type", cty.StringVal("c5.2xlarge"))
+	if !useExistingKeyPair {
+		awsInstanceBody.SetAttributeTraversal("key_name", hcl.Traversal{
+			hcl.TraverseRoot{
+				Name: "aws_key_pair",
+			},
+			hcl.TraverseAttr{
+				Name: "kp",
+			},
+			hcl.TraverseAttr{
+				Name: "key_name",
+			},
+		})
+	} else {
+		awsInstanceBody.SetAttributeValue("key_name", cty.StringVal(existingKeyPairName))
+	}
+	var securityGroupList []cty.Value
+	securityGroupList = append(securityGroupList, cty.StringVal(securityGroupName))
+	awsInstanceBody.SetAttributeValue("security_groups", cty.ListVal(securityGroupList))
+}
+
+func setOutput(rootBody *hclwrite.Body) {
+	outputEip := rootBody.AppendNewBlock("output", []string{"instance_eip"})
+	outputEipBody := outputEip.Body()
+	outputEipBody.SetAttributeTraversal("value", hcl.Traversal{
+		hcl.TraverseRoot{
+			Name: "aws_eip",
+		},
+		hcl.TraverseAttr{
+			Name: "myeip",
+		},
+		hcl.TraverseAttr{
+			Name: "public_ip",
+		},
+	})
+
+	outputInstanceID := rootBody.AppendNewBlock("output", []string{"instance_id"})
+	outputInstanceIDBody := outputInstanceID.Body()
+	outputInstanceIDBody.SetAttributeTraversal("value", hcl.Traversal{
+		hcl.TraverseRoot{
+			Name: "aws_instance",
+		},
+		hcl.TraverseAttr{
+			Name: "fuji_node[0]",
+		},
+		hcl.TraverseAttr{
+			Name: "id",
+		},
+	})
+}
+
+func createAnsibleHostInventory(inventoryPath, elasticIP, certFilePath string) error {
+	if err := os.MkdirAll(inventoryPath, os.ModePerm); err != nil {
+		log.Fatal(err)
+	}
+	inventoryHostsFile := inventoryPath + "/hosts"
+	inventoryFile, err := os.Create(inventoryHostsFile)
+	if err != nil {
+		return err
+	}
+	alias := "aws-node "
+	elasticIPToUse := elasticIP[1 : len(elasticIP)-2]
+	alias += "ansible_host="
+	alias += elasticIPToUse
+	alias += " ansible_user=ubuntu "
+	alias += fmt.Sprintf("ansible_ssh_private_key_file=%s", certFilePath)
+	alias += " ansible_ssh_common_args='-o StrictHostKeyChecking=no'"
+	_, err = inventoryFile.WriteString(alias + "\n")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func runTerraform() (string, string, error) {
+	var instanceID string
+	var elasticIP string
+	if err := exec.Command("terraform", "init").Run(); err != nil {
+		return "", "", err
+	}
+	var stdBuffer bytes.Buffer
+	cmd := exec.Command("terraform", "apply", "-auto-approve")
+	mw := io.MultiWriter(os.Stdout, &stdBuffer)
+	cmd.Stdout = mw
+	cmd.Stderr = mw
+	if err := cmd.Run(); err != nil {
+		return "", "", err
+	}
+
+	instanceIDOutput, err := exec.Command("terraform", "output", "instance_id").Output()
+	if err != nil {
+		return "", "", err
+	}
+	instanceID = string(instanceIDOutput)
+	eipOutput, err := exec.Command("terraform", "output", "instance_eip").Output()
+	if err != nil {
+		return "", "", err
+	}
+	elasticIP = string(eipOutput)
+	return instanceID, elasticIP, nil
+}
+
+func handleCerts(certName string) error {
+	err := os.Chmod(certName, 0o400)
```

Comment:
if certName a path?

Reply:
can we change the param name and also make it clear in caller code because it seems is only cert name

Reply:
I continue not understanding why a certName is passed onto os.Chmod as if it were a path